react/src/components/FolderCreateModal.tsx

```diff
@@ -224,8 +240,13 @@ const FolderCreateModal: React.FC<FolderCreateModalProps> = ({
           style={{ flex: 1, marginBottom: 0 }}
         >
           <Radio.Group>
-            <Radio value={'user'}>User</Radio>
-            <Radio value={'project'}>Project</Radio>
+            {_.includes(allowedTypes, 'user') ? (
+              <Radio value={'user'}>User</Radio>
+            ) : null}
+            {(userRole === 'admin' || userRole === 'superadmin') &&
+            _.includes(allowedTypes, 'group') ? (
+              <Radio value={'project'}>Project</Radio>
+            ) : null}
           </Radio.Group>
```

Comment:
I have a question. Why do we need to check both `role` and `allowedTypes`? Could you provide more details? It would be helpful to add a comment about this.